### PR TITLE
feat: use class path string as custom logit processor parameter

### PR DESCRIPTION
**Pull request by pengye91**
## Motivation

1. For built in custom logit processors, instead of importing from sglang package, which requires `sglang` as a dependency for clients, use class path string as the extra parameter.

now we can use the `custom_logit_processor` parameter in client codes without `import sglang` like this:
```python
import openai
from rich.pretty import pprint

client = openai.Client(base_url="http://127.0.0.1:30000/v1", api_key="*")
response = client.chat.completions.create(
    model="deepseek-ai/DeepSeek-R1",
    messages=[
        {
            "role": "user",
            "content": "Question: Is Paris the Capital of France?",
        }
    ],
    max_tokens=1024,
    extra_body={
        "custom_logit_processor": "sglang.srt.sampling.custom_logit_processor:DeepSeekR1ThinkingBudgetLogitProcessor",
        "custom_params": {
            "thinking_budget": 512,
        },
    },
)
pprint(response)
```

2. Raise a `ValueError` at request initialization if the passed `custom_logit_processor` parameter is not valid, instead of raising a `RuntimeError` which will directly kill the server.
3. Add a `validate` method in `GenerateReqInput` class for all possible validation process.

## Modifications

1. Added a `validate` method in `GenerateReqInput` class for all possible validation process.
2. Added `_validate_custom_logit_processor` method in `GenerateReqInput` class to validate if the `custom_logit_processor` parameter is valid.
3. Added a `from_class_path` method in `CustomLogitProcessor` class to load the logit processor from class path string.
5. Keep the old custom logit processor loading method for compatibility and for not built-in logit processors.
6. Added related test cases.
7. Added related docs.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-style-guidance).
- [x] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/sampling/custom_logit_processor.py [hunk 1]
```diff
@@ -20,6 +21,20 @@ def _cache_from_str(json_str: str):
     return dill.loads(bytes.fromhex(data["callable"]))
 
 
+@lru_cache(maxsize=None)
+def _cache_from_class_path(class_path: str):
+    """Get the processor directly from the class path."""
+    class_path_split_list = class_path.split(":")
```
Comment (CatherineSue) [inline python/sglang/srt/sampling/custom_logit_processor.py:27]: Any reason we use `:` as separator?

**Review by CatherineSue — COMMENTED** [review]
overall LGTM. Might need another peer review from experts. I can approve once CI passes

File: python/sglang/srt/sampling/custom_logit_processor.py (hunk 1 above)
Comment (pengye91) [reply python/sglang/srt/sampling/custom_logit_processor.py:27]: I'm following a common pattern for WSGI applications, like `myapp.wsgi:application`, and `uvicorn run module.path:app_object`, and even [vllm logit processor initialization method](https://docs.vllm.ai/en/latest/features/custom_logitsprocs/#method-1-pass-the-custom-logits-processor-fully-qualified-class-name-fqcn-to-vllm-at-initialization-time), but I can change it to dot separator if that's compatible with SGlang's code style.

**Comment by acelyc111** [comment]
Hi @pengye91, is this work on going? There is a CVE which this PR can resolve https://antiproof.ai/blog/three-rces-in-sglang/, we can push it forward.

File: python/sglang/srt/managers/io_struct.py [hunk 2]
```diff
@@ -256,6 +257,11 @@ def normalize_batch_and_arguments(self):
         else:
             self._normalize_batch_inputs()
 
+    def validate(self):
+        """Validate the request parameters."""
+        self._validate_custom_logit_processor()
+        # TODO: Add more validations as needed
+
     def _validate_inputs(self):
```
Comment (acelyc111) [inline python/sglang/srt/managers/io_struct.py:265]: How about moving these `_validate_*` into the newly added `validate()`?
